packages/camera/camera_android_camerax/android/src/main/java/io/flutter/plugins/camerax/InstanceManager.java

```diff
@@ -107,30 +126,38 @@ public Long getIdentifierForStrongReference(Object instance) {
   /**
    * Adds a new instance that was instantiated from Dart.
    *
-   * <p>If an instance or identifier has already been added, it will be replaced by the new values.
-   * The Dart InstanceManager is considered the source of truth and has the capability to overwrite
-   * stored pairs in response to hot restarts.
+   * <p>The same instance can be added multiple times, but each identifier must be unique. This
+   * allows two objects that are equivalent (e.g. the `equals` method returns true and their
+   * hashcodes are equal) to both be added.
+   *
+   * <p>If the manager is closed, the addition is ignored and a warning is logged.
    *
    * @param instance the instance to be stored.
-   * @param identifier the identifier to be paired with instance. This value must be >= 0.
+   * @param identifier the identifier to be paired with instance. This value must be >= 0 and
+   *     unique.
    */
   public void addDartCreatedInstance(Object instance, long identifier) {
-    assertManagerIsNotClosed();
+    if (isClosed()) {
+      Log.w(TAG, CLOSED_WARNING);
+      return;
+    }
     addInstance(instance, identifier);
   }
 
   /**
    * Adds a new instance that was instantiated from the host platform.
    *
-   * <p>If an instance has already been added, this will replace it. {@code #containsInstance} can
-   * be used to check if the object has already been added to avoid this.
-   *
-   * @param instance the instance to be stored.
-   * @return the unique identifier stored with instance.
+   * @param instance the instance to be stored. This must be unique to all other added instances.
+   * @return the unique identifier stored with instance. If the manager is closed, returns -1.
    */
   public long addHostCreatedInstance(Object instance) {
-    assertManagerIsNotClosed();
-
+    if (isClosed()) {
+      Log.w(TAG, CLOSED_WARNING);
+      return -1;
```

Comment:
Consider making this a constant int. 

For example 
```
// Constant returned from #addHostCreatedInstance() if the manager is closed. 
public static final int INSTANCE_CLOSED = -1;

public long addHostCreatedInstance(Object instance) {
    if (isClosed()) {
      Log.w(TAG, CLOSED_WARNING);
      return INSTANCE_CLOSED;
}
```

This makes code that uses this method more readable by bringing context to the value check without having to go read the documentation.

Reply:
@bparrishMines just fyi, I'm going to make this change